### PR TITLE
Extended publishing of libraries to npm

### PR DESCRIPTION
**Pull request by jacogr**
Publish all external libraries now, `parity.js`, `etherscan` & `shapeshift` integrations

### CODE REVIEW

File: js/npm/parity/README.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
-# parity.js
+# @parity/parity.js
 
 Parity.js is a thin, fast, Promise-based wrapper around the Ethereum APIs.
 
+[https://github.com/ethcore/parity/js/src/api](https://github.com/ethcore/parity/js/src/api)
```
Comment (ngotchac) [inline js/npm/parity/README.md]: This should be `https://github.com/ethcore/parity/tree/master/js/src/api`

File: js/npm/etherscan/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# @parity/Etherscan
+
+A thin, lightweight promise wrapper for the api.etherscan.io/apis service, exposing a common endpoint for use in JavaScript applications.
+
+[https://github.com/ethcore/parity/js/src/3rdparty/etherscan](https://github.com/ethcore/parity/js/src/3rdparty/etherscan)
```
Comment (ngotchac) [inline js/npm/etherscan/README.md]: This should be `https://github.com/ethcore/parity/tree/master/js/src/3rdparty/etherscan`
Plus, don't we want to add the README there, so users can read it navigating to this URL ?

File: js/npm/parity/package.json [hunk 3]
```diff
@@ -26,8 +26,8 @@
   "devDependencies": {
   },
   "dependencies": {
-    "bignumber.js": "^2.3.0",
-    "js-sha3": "^0.5.2",
-    "node-fetch": "^1.6.3"
+    "bignumber.js": "~2.3.0",
```
Comment (ngotchac) [inline js/npm/parity/package.json]: Why not using `2.3.0` directly? I'm sure the patches won't break anything, but we never know...

File: js/npm/shapeshift/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# @parity/ShapeShift
+
+A thin ES6 promise wrapper around the shapeshift.io APIs as documented at https://shapeshift.io/api
+
+[https://github.com/ethcore/parity/js/src/3rdparty/shapeshift](https://github.com/ethcore/parity/js/src/3rdparty/shapeshift)
```
Comment (ngotchac) [inline js/npm/shapeshift/README.md]: `https://github.com/ethcore/parity/tree/master/js/src/3rdparty/shapeshift` + same comment on README as per etherscan

File: js/npm/parity/package.json (hunk 3 above)
Comment (jacogr) [inline js/npm/parity/package.json]: It is actually bad form doing what we are doing. We should be using shinkwrap. (Extra step, but right approach)

File: js/npm/shapeshift/README.md (hunk 4 above)
Comment (jacogr) [inline js/npm/shapeshift/README.md]: Updated.

File: js/npm/etherscan/README.md (hunk 2 above)
Comment (jacogr) [inline js/npm/etherscan/README.md]: Updated.

As for the README, do agree, but actually need to solve this with the source in npm issue as well. Will log and revisit.

**Comment by ngotchac** [comment]
I added some tests to the published libraries. For `parity` it is juste a smoke one, testing the the object exists and that it contains `Api` and `Abi`.
For `etherscan` and `shapeshift`, it copies the already existing tests (slightly modified) into the published directory.

I added a call to `npm test` from each published library in the `dryrun-npm.sh` script, but this might be added in another non dry-run script.

If this is useless, feel free to revert my commits (`02c2dc0` and `861280b`)

**Pull request merged**